front/src/applications/stdcm/hooks/useProjectedTrainsForStdcm.ts

```diff
@@ -53,14 +102,35 @@ const useProjectedTrainsForStdcm = (stdcmResponse?: StdcmSuccessResponse) => {
   });
 
   useEffect(() => {
-    const newSpaceTimeData = Array.from(projectedTimetableTrainsById.values());
+    if (stdcmResponse) {
+      const relevantTrainScheduleIds = keepTrainsRunningDuringStdcm(
+        stdcmResponse,
+        trainScheduleSummariesById
+      );
```

Comment:
Could we maybe extract this as a `useMemo()`? That way we don't have to duplicate it in both `useEffect()` calls.

Reply:
Hm, sounds like that wouldn't work well since we omit dependencies in `useEffect()`…